### PR TITLE
Set Shim OOM score to +1 of containerd

### PR DESCRIPTION
**Pull request by praveensastry**
*Issue #, if available:* #217 

*Description of changes:* Added a method to get the OOM score for containerd process and increment the score by 1 to set shim OOM score

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by praveensastry** [comment]
@sipsma Would be great if you can take a look and provide feedback.

File: firecracker-control/local.go [hunk 1]
```diff
@@ -338,3 +338,15 @@ func (s *local) newShim(ns, vmID, containerdAddress string, shimSocket *net.Unix
 func isEADDRINUSE(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "address already in use")
 }
+
+func setShimOOMScore(pid int) error {
+	score, err := sys.GetOOMScoreAdj(pid)
```
Comment (sipsma) [inline firecracker-control/local.go]: Here we want to get the OOM score of the containerd process, but I think it's currently getting the OOM score of the shim child process instead.

This firecracker-control code runs within the containerd process itself, so to get the containerd pid all we have to do is call [`os.Getpid()`](https://golang.org/pkg/os/#Getpid). Then we can plug that into `sys.GetOOMScoreAdj`.

Comment (praveensastry) [reply firecracker-control/local.go]: Thanks for clarifying that! I thought to do the same when I was looking at containerd's implementation but got confused when I was implementing it.

Comment (sipsma) [reply firecracker-control/local.go]: Ah okay, one more clarification. Your update is correct in that we are now using containerd's pid to get the OOM score. However, we actually want to set the OOM score on the pid you were using previously (`cmd.Process.Pid`). The end result we want is that `cmd.Process.Pid` gets the OOM score +1 of containerd's OOM score.

This is a little different than what containerd's code does, so I understand the confusion :-)

**Comment by praveensastry** [comment]
Travis build failed due to an environmental issue (I believe) with this error message `The command "travis_setup_go" failed and exited with 86 during .`

**Comment by samuelkarp** [comment]
@praveensastry I've rerun the Travis job for you

**Comment by praveensastry** [comment]
Thank you @sipsma for the opportunity. Looking forward to contributing more :)

**Pull request merged**